### ISSUE TITLE
[Bug]: ListKeymap Uncaught RangeError: Position -1 out of range

### ISSUE THREAD

**Issue by adjourn** [ueberdosis/tiptap#4368]
### Which packages did you experience the bug in?

Isolated to only list-keymap, happens in tiptap.dev example as well

### What Tiptap version are you using?

2.1.6

### What’s the bug you are facing?

Steps to reproduce:
* Open https://tiptap.dev/api/extensions/list-keymap, scroll to live editor
* Add bullet list (item) as first block
* Add plain paragraph as second block
* Select all and press `Backspace`

Get the error: `Uncaught RangeError: Position -1 out of range`.
Content is removed and editor seems to be usable after the error without having to reload.

Observations:
* Doesn't happen with bullet nor ordered list if `list-keymap` is not present
* Described issue occurs only if there's 1 bullet list item
* Doesn't happen when you press `Delete` key
* Bullet list (item) nor following paragraph need to contain any content, error is thrown even if both are empty
* Don't see the issue if bullet list (item) is not the first block in the document

---

#### Just mentioning, possibly related/same cause:
Same situation as before, only difference is that there are >1 bullet list items

Issue: last list item is removed and content is shuffled around.

```html
Starting doc:
<ul>
  <li>A</li>
  <li>B</li>
</ul>
<p>C</p>

*Select all and hit Backspace*

End up with doc:
<ul>
  <li>BA</li>
</ul>
```

### What browser are you using?

Chrome

### Code example

https://tiptap.dev/api/extensions/list-keymap

### What did you expect to happen?

Paragraph and bullet list are deleted and replaced with empty enforced paragraph.

### Did you update your dependencies?

- [X] Yes, I’ve updated my dependencies to use the latest version of all packages.

### Are you sponsoring us?

- [ ] Yes, I’m a sponsor. 💖

**Comment by bdbch** [comment]
Good find. Guess we miss a check if a node range is selected where if it is we should fall back to the default backspace functionality.

**Comment by juraj98** [comment]
I looked into this, and there is more issues with this behavior with the extension. For example if the content looks like this:

```
- This is first list item
- This is second list item
This is paragraph
```

and we select all and hit backspace, the result will be this:

```
- This is first list itemThis is second list item
This is paragraph
```

--- 

But after playing with the code for a while, I figured that just checking if the selection is a range (not sure if this is correct terminology but `if (from !== to)`), we can rely on default backspace behavior and the result will always (I hope) be what's expected. Will create PR in a moment.
